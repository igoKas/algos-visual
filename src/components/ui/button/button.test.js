import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

describe('Button test', () => {
    it('render button with text', () => {
        const button = renderer
        .create(<Button text='text' />)
        .toJSON();
        expect(button).toMatchSnapshot();
    })

    it('render button without text', () => {
        const button = renderer
        .create(<Button text='' />)
        .toJSON();
        expect(button).toMatchSnapshot();
    })

    it('render disabled button', () => {
        const button = renderer
        .create(<Button disabled={true} />)
        .toJSON();
        expect(button).toMatchSnapshot();
    })

    it('render button with loader', () => {
        const button = renderer
        .create(<Button isLoader={true} />)
        .toJSON();
        expect(button).toMatchSnapshot();
    })

    it('onClick has been called', () => {
        window.alert = jest.fn();
        render(<Button text='text' onClick={() => alert('onClick has been called')} />)
        const link = screen.getByText('text');

        fireEvent.click(link);

        expect(window.alert).toHaveBeenCalledWith('onClick has been called');
    })
})