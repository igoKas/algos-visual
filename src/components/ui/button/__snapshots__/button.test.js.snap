// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test render button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button test render button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-cy="text"
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button test render button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-cy=""
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Button test render disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
